File: androidApp/src/main/java/com/example/justdesserts/androidApp/models/Dessert.kt
package com.example.justdesserts.androidApp.models


data class Dessert(
    val action: DessertAction = DessertAction.READ,
    val dessertId: String,
    val name: String,
    val description: String,
    val imageUrl: String,
    val reviews: List<Review> = emptyList()
) {
    fun toQueryString(): String {
        return "?id=${this.dessertId}&name=${this.name}&description=${this.description}&imageUrl=${this.imageUrl}"
    }
}

File: iosApp/iosApp/Features/Desserts/Models/Dessert.swift
//
//  Dessert.swift
//  iosApp
//
//  Created by Michael Stromer on 1/1/21.
//  Copyright © 2021 orgName. All rights reserved.
//

import Foundation
import shared

struct Dessert {
    let action: DessertAction
    let dessertId: String
    let name: String
    let description: String
    let imageUrl: String
    let reviews: [Review]
    
    init(action: DessertAction = .READ, dessertId: String, name: String = "", description: String = "", imageUrl: String = "", reviews: [Review] = []) {
        self.action = action
        self.dessertId = dessertId
        self.name = name
        self.description = description
        self.imageUrl = imageUrl
        self.reviews = reviews
    }
}
